### ISSUE TITLE
Epic - Website build and deployment separation

### ISSUE THREAD

**Issue by sean1588** [pulumi/docs#9829]
Our content can take up to 1 hour and 45 min to deploy once merged into master in the pulumi-hugo repo. Separating these builds will enable us to speed up the deployment time of the marketing and docs content. This will also address one of the pitfalls of our system, being a single point of failure where often times our deployment pipeline can end up being blocked by build failures in registry. This requires a significant amount of time addressing registry failures in docs-ops to make sure our pipeline is always clear of any failures so we don’t block marketing content from going out, which can be a concern when time-sensitive content needs to go live.

Epic to track build separation workstream:

**Phase 1** - configure the registry to deploy independently. This requires the creation of a new pulumi program dedicated to managing the registry's infrastructure. Scripts will be configured to generate registry documentation and build pages. Additionally, the registry will be equipped with its own cloudfront distribution, serving as the origin for the /registry route. The cloudfront CDN URL will be exported as a stack output, enabling it to be referenced as an origin in the docs program. 
https://github.com/pulumi/registry/pull/2963 (done in testing)
https://github.com/pulumi/registry/pull/3150 (PR for prod)

**Phase 2** - Configure docs to do partial deploy: configuring the docs repository to perform a partial deploy, excluding the registry, since this will all be done in the registry repository now.  
https://github.com/pulumi/docs/pull/9619 (done in testing)
https://github.com/pulumi/docs/pull/9853 (PR for prod)

**Phase 3** - Consolidate pulumi-hugo content into docs: involves migrating the content from the pulumi-hugo repository to the docs repository, subsequently archiving the pulumi-hugo repository. All contributions to marketing and docs will then be done here instead of the docs repo.

**Comment by cnunciato** [comment]
This work is complete.
